test/e2e/density.go

```diff
@@ -231,9 +235,38 @@ var _ = Describe("Density", func() {
 		}
 		itArg := testArg
 		It(name, func() {
+			uuid := string(util.NewUUID())
 			totalPods := itArg.podsPerMinion * minionCount
-			RCName = "my-hostname-density" + strconv.Itoa(totalPods) + "-" + string(util.NewUUID())
+			nameStr := strconv.Itoa(totalPods) + "-" + uuid
+			ns = "e2e-density" + nameStr
+			RCName = "my-hostname-density" + nameStr
 			RunRC(c, RCName, ns, "gcr.io/google_containers/pause:go", totalPods)
+			By("waiting for all events to be recorded")
+			last := -1
+			current := 0
+			var events *api.EventList
+			for last < current {
+				e, err := c.Events(ns).List(
+					labels.Everything(),
+					fields.Set{
+						"involvedObject.namespace": ns,
+					}.AsSelector(),
+				)
+				expectNoError(err)
+				last = current
```

Comment:
#6546 should hopefully merge today. I am concerned that this method of getting events is going to be hard enough on apiserver that it will actually cause a performance problem. I'm also not sure about this stopping condition-- is it a guarantee that no events are generated in steady state? That may currently be the case but I don't think it's guaranteed...